extensions/wikia/WikiFactory/Close/gcs_bucket_remover.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+use Wikia\Logger\Loggable;
+
+class GcsBucketRemover {
+	use Loggable;
+
+	public function remove( int $wikiId ) {
+		try {
+			$wgUploadPath = WikiFactory::getVarValueByName( 'wgUploadPath', $wikiId );
+
+			if ( empty( $wgUploadPath ) ) {
+				$this->info( "Upload path path is empty, leave early\n" );
+
+				return false;
+			}
+
+			/** @var GcsFileBackend $backend */
+			$backend = FileBackendGroup::singleton()->get( 'gcs-backend' );
+
+			$this->info( sprintf( 'Removing images from path %s', $wgUploadPath ) );
+			$path = trim( parse_url( $wgUploadPath, PHP_URL_PATH ), '/' );
+
+			$this->info( sprintf( 'Getting file list for %s', $path ) );
+
+			$path = StringUtils::escapeRegexReplacement( "mwstore://{$backend->getName()}/" . $path );
+
+			$objects = iterator_to_array( $backend->getFileList( [ 'dir' => $path ] ) );
```

Comment:
Won't this blow up with an OOM exception if there are too many images?

Reply:
I will try to revert this - afaik current implementation does similar thing, so I wasn't concerned about OOM. I did it during debugging - this iterator implementation could be incorrect - it returned 1 from count with no objects inside, but we don't need it after lines below are removed.